aptos-move/move-examples/veiled_coin/sources/veiled_coin.move

```diff
@@ -0,0 +1,670 @@
+	x5: RistrettoPoint,
+	alpha1: Scalar,
+	alpha2: Scalar,
+	alpha3: Scalar,
+	alpha4: Scalar,
+    }
+
+    /// Deserializes and returns a VeiledWithdrawalProof given its byte representation
+    fun deserialize_withdrawal_proof<CoinType>(proof_bytes: vector<u8>): Option<VeiledWithdrawalProof<CoinType>> {
+	assert!(vector::length<u8>(&proof_bytes) == 288, EBYTES_WRONG_LENGTH);
+	let x1_bytes = vector::trim<u8>(&mut proof_bytes, 32);
+	let x1 = ristretto255::new_point_from_bytes(x1_bytes);
+	if (!std::option::is_some<RistrettoPoint>(&x1)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let x1 = std::option::extract<RistrettoPoint>(&mut x1);
+
+	let x2_bytes = vector::trim(&mut proof_bytes, 32);	
+	let x2 = ristretto255::new_point_from_bytes(x2_bytes);
+	if (!std::option::is_some<RistrettoPoint>(&x2)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let x2 = std::option::extract<RistrettoPoint>(&mut x2);	
+
+	let x3_bytes = vector::trim(&mut proof_bytes, 32);	
+	let x3 = ristretto255::new_point_from_bytes(x3_bytes);
+	if (!std::option::is_some<RistrettoPoint>(&x3)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let x3 = std::option::extract<RistrettoPoint>(&mut x3);
+
+	let x4_bytes = vector::trim(&mut proof_bytes, 32);	
+	let x4 = ristretto255::new_point_from_bytes(x4_bytes);
+	if (!std::option::is_some<RistrettoPoint>(&x4)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let x4 = std::option::extract<RistrettoPoint>(&mut x4);
+
+	let x5_bytes = vector::trim(&mut proof_bytes, 32);	
+	let x5 = ristretto255::new_point_from_bytes(x5_bytes);
+	if (!std::option::is_some<RistrettoPoint>(&x5)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let x5 = std::option::extract<RistrettoPoint>(&mut x5);
+
+	let alpha1_bytes = vector::trim(&mut proof_bytes, 32);
+	let alpha1 = ristretto255::new_scalar_from_bytes(alpha1_bytes);
+	if (!std::option::is_some(&alpha1)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let alpha1 = std::option::extract(&mut alpha1);
+
+	let alpha2_bytes = vector::trim(&mut proof_bytes, 32);
+	let alpha2 = ristretto255::new_scalar_from_bytes(alpha2_bytes);
+	if (!std::option::is_some(&alpha2)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let alpha2 = std::option::extract(&mut alpha2);
+
+	let alpha3_bytes = vector::trim(&mut proof_bytes, 32);
+	let alpha3 = ristretto255::new_scalar_from_bytes(alpha3_bytes);
+	if (!std::option::is_some(&alpha3)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let alpha3 = std::option::extract(&mut alpha3);
+
+	let alpha4_bytes = vector::trim(&mut proof_bytes, 32);
+	let alpha4 = ristretto255::new_scalar_from_bytes(alpha4_bytes);
+	if (!std::option::is_some(&alpha4)) {
+	    return std::option::none<VeiledWithdrawalProof<CoinType>>()
+	};
+	let alpha4 = std::option::extract(&mut alpha4);
+
```

Comment:
Check here that the left-over `proof_bytes` are of size 0 and return `None` if they are not.

Reply:
We check at the start of the function that the total proof bytes are 288 so there shouldn't be any left over